storage/range.go

```diff
@@ -235,13 +235,35 @@ func (r *Range) String() string {
 	return fmt.Sprintf("range=%d (%s-%s)", r.Desc().RaftID, r.Desc().StartKey, r.Desc().EndKey)
 }
 
-// start begins by gossiping the cluster ID (only happens if the range
-// is the first in the key space). This is necessary to get the cluster
-// started from scratch.
+// start begins by gossiping the configs. It also gossips the sentinel if we're
+// the first range (the store will do this regularly, but this is a good time
+// to get started quickly).
 func (r *Range) start() {
-	if r.IsFirstRange() {
-		r.startGossip(r.rm.Stopper())
+	// Being the first range comes with extra responsibility for the gossip
+	// sentinel and first range metadata.
+	// TODO(tschottdorf) really want something more streamlined, such as:
+	//
+	// if r.rm.Stopper().StartTask() {
+	// 	go func() {
+	// 		r.maybeGossipFirstRangeWithLease()
+	// 		r.rm.Stopper().FinishTask()
+	// 	}()
+	// }
+	//
+	// but
+	// a) in some tests that goroutine gets stuck acquiring the lease, probably
+	//    due to Raft losing the command
```

Comment:
I definitely see the lease acquiration deadlock waiting for something to come back on the command channel, even though it's run in a goroutine.  You're right, since everything should end up in MultiNode's `g.pending` map at some point, it should be reproposed after a successful election. Maybe it's in a test that doesn't properly start everything.